src/ui/public/vis_maps/heatmap.js

```diff
@@ -0,0 +1,200 @@
+import L from 'leaflet';
+import _ from 'lodash';
+import d3 from 'd3';
+import { EventEmitter } from 'events';
+
+/**
+ * Map overlay: canvas layer with leaflet.heat plugin
+ *
+ * @param map {Leaflet Object}
+ * @param geoJson {geoJson Object}
+ * @param params {Object}
+ */
+export default class Heatmap extends EventEmitter {
```

Comment:
The heatmap still has this empty legend highlighted by the red rectangle below:

<img width="1229" alt="screen shot 2017-03-24 at 8 30 18 am" src="https://cloud.githubusercontent.com/assets/627123/24294293/4134e9c2-106c-11e7-98f2-e07cfa10af26.png">